nmostesting/suites/IS0401Test.py

```diff
@@ -1556,6 +1563,277 @@ def test_26(self, test):
+        """Node API implements BCP-004-01 Receiver Capabilities constraint set preferences"""
+        return self.do_test_constraint_set_meta(test, "preference", "preferences")
+
+    def test_27_6(self, test):
+        """Node API implements BCP-004-01 Receiver Capabilities enabled/disabled constraint sets"""
+        return self.do_test_constraint_set_meta(test, "enabled", "enabled/disabled flags")
+
+    def do_test_constraint_set_meta(self, test, meta, description, warn_not_all=False):
+        api = self.apis[RECEIVER_CAPS_KEY]
+
+        receivers_valid, receivers_response = self.do_request("GET", self.node_url + "receivers")
+
+        no_receivers = True
+        no_constraint_sets = True
+        no_meta = True
+        all_meta = True
+
+        if receivers_valid and receivers_response.status_code == 200:
+            try:
+                for receiver in receivers_response.json():
+                    no_receivers = False
+                    if "constraint_sets" in receiver["caps"]:
+                        no_constraint_sets = False
+                        for constraint_set in receiver["caps"]["constraint_sets"]:
+                            if "urn:x-nmos:cap:meta:" + meta in constraint_set:
+                                no_meta = False
+                            else:
+                                all_meta = False
+            except json.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from Node API")
+            except KeyError as e:
+                return test.FAIL("Unable to find expected key in the Receiver: {}".format(e))
+
+        if no_receivers:
+            return test.UNCLEAR("No Receivers were found on the Node")
+        elif no_constraint_sets:
+            return test.OPTIONAL("No BCP-004-01 'constraint_sets' were identified in Receiver caps",
+                                 "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                 "/docs/1.0._Receiver_Capabilities.html#listing-constraint-sets"
+                                 .format(api["spec_branch"]))
+        elif no_meta:
+            return test.OPTIONAL("No BCP-004-01 'constraint_sets' have {}".format(description),
+                                 "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                 "/docs/1.0._Receiver_Capabilities.html#constraint-set-{}"
+                                 .format(api["spec_branch"], meta))
+        elif warn_not_all and not all_meta:
+            return test.WARNING("Only some BCP-004-01 'constraint_sets' have {}".format(description),
+                                "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                "/docs/1.0._Receiver_Capabilities.html#constraint-set-{}"
+                                .format(api["spec_branch"], meta))
+        else:
+            return test.PASS()
+
+    def test_27_7(self, test):

```

Comment:
This test results in a pass if there are no `constraint_sets` present. It might be worth an 'optional' response to match the others?